### PR TITLE
React on Rails ReasonHelloWorld component

### PR DESCRIPTION
**Pull request by ljuster**
**REASON/REACT**   
    - bsconfig
    - bsb make and clean commands in build
    - ReasonHelloWorld.re component
    - added component to client registration file
  **RAILS**
    - add route
    - add hyperlink on main page
    - view file

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1215)

### CODE REVIEW

File: spec/dummy/client/app/startup/clientRegistration.jsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import 'es5-shim';
 import ReactOnRails from 'react-on-rails';
 
 import HelloWorld from '../components/HelloWorld';
+import HelloWorldReason from '../components/HelloWorldReason.bs';
```
Comment (ljuster) [inline spec/dummy/client/app/startup/clientRegistration.jsx:7]: @justin808 as is, it won't recognize `HelloWorldReason.bs` component. 
```
clientRegistration.jsx?209f:1 Uncaught Error: Cannot find module "../components/HelloWorldReason.bs"
```

Comment (justin808) [reply spec/dummy/client/app/startup/clientRegistration.jsx:7]: This is just not how you integrate the bs into the JS. The JS code needs the compiled file to import.

**Comment by justin808** [comment]
@ljuster how's this going? We need CI to pass.

**Comment by justin808** [comment]
@Judahmeek check this one out. Would be great to get this merged.

**Comment by justin808** [comment]
@Judahmeek @behraaangm can we get this merged into spec/dummy?

**Comment by behraaangm** [comment]
Hey @justin808, I wanna make a new branch from this work.
the `package.json` in spec/dummy/client is should be removed, right?

**Comment by justin808** [comment]
> Hey @justin808, I wanna make a new branch from this work.

Sure!

> the package.json in spec/dummy/client is should be removed, right?

Yes!

**Comment by justin808** [comment]
See #1399

**Pull request closed without merging**